engine/order_manager.go

```diff
@@ -743,27 +832,45 @@ func (s *store) modifyExisting(id string, mod *order.Modify) error {
 
 // upsert (1) checks if such an exchange exists in the exchangeManager, (2) checks if
 // order exists and updates/creates it.
-func (s *store) upsert(od *order.Detail) error {
+func (s *store) upsert(od *order.Detail) (resp *OrderUpsertResponse, err error) {
+	if od == nil {
+		return &OrderUpsertResponse{}, errNilOrder
```

Comment:
`return nil, errNilOrder` I've read https://github.com/thrasher-corp/gocryptotrader/pull/775#discussion_r702890047 your comment on this, but when you're also returning an error, you only ever check that and return, so no need to instantiate a new struct+pointer

Reply:
Ah yes I was confused